OmsAgent/watcherutil.py

```diff
@@ -162,3 +251,261 @@ def watch(self):
             time.sleep(60 * 5)
 
         pass
+
+    def monitor_heartbeat(self, self_mon_info, log_file_marker):
+        """
+            Monitor heartbeat health. OMS output plugin will update the timestamp
+            of new heartbeat file every 5 minutes. We will check if it is updated
+            If not, we will look into omsagent logs and look for specific error logs
+            which indicate we are in non recoverable state.             
+        """
+        heartbeat_file = '/var/opt/microsoft/omsagent/log/heartbeat'
```

Comment:
The ODSIngestion.status file is already monitoring heartbeat indirectly since it uses the same channel. There isn't any need for adding another file just for the heartbeat,

Reply:
I want a signal which explicitly says that hertbeat is being emitted periodically. ODSIngestion status file does not tell me that with confidence. It can be updated when any other event is ingested. 
For scope of this change i need definiteve signal saying heartbeat is flowing end to end.

Reply:
Are you adding this file in the input plugin? If so, this is fine.